pkg/controllers/cluster/cluster_controller.go

```diff
@@ -321,6 +326,40 @@ func (c *Controller) ExecutionSpaceExistForCluster(clusterName string) (bool, er
 	return true, nil
 }
 
+func (c *Controller) deleteFinalizerForWorks(cluster *clusterv1alpha1.Cluster, workSpace *corev1.Namespace) error {
+	//delete finalizers of work objects When the sync-mode is pull mode and cluster status is notready or unknown state
+	if cluster.Spec.SyncMode == clusterv1alpha1.Pull && !util.IsClusterReady(&cluster.Status) {
```

Comment:
Suggest to put the `if` at the function out, this function `deleteFinalizerForWorks` focus on action.

Like
```
if cluster.Spec.SyncMode == clusterv1alpha1.Pull && !util.IsClusterReady(&cluster.Status) {
    if err := c.deleteFinalizerForWorks(cluster, executionSpaceObj); err != nil {
		klog.Errorf("Error while deleting finalizers of work which in %s: %s", executionSpaceName, err)
		return err
	}
}
```

Reply:
fix it